pkg/vmnetworkscontroller/vm_controller.go

```diff
@@ -0,0 +1,67 @@
+package vmnetworkscontroller
+
+import (
+	"context"
+
+	"github.com/go-logr/logr"
+
+	"k8s.io/apimachinery/pkg/runtime"
+
+	controllerruntime "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/event"
+	"sigs.k8s.io/controller-runtime/pkg/predicate"
+
+	virtv1 "kubevirt.io/api/core/v1"
+)
+
+// VirtualMachineReconciler reconciles a VirtualMachine object
+type VirtualMachineReconciler struct {
+	client.Client
+	Log     logr.Logger
+	Scheme  *runtime.Scheme
+	manager controllerruntime.Manager
+}
+
+func NewVMReconciler(manager controllerruntime.Manager) *VirtualMachineReconciler {
+	return &VirtualMachineReconciler{
+		Client:  manager.GetClient(),
+		Log:     controllerruntime.Log.WithName("controllers").WithName("VirtualMachine"),
+		Scheme:  manager.GetScheme(),
+		manager: manager,
+	}
+}
+
+func (r *VirtualMachineReconciler) Reconcile(
+	ctx context.Context,
+	request controllerruntime.Request,
+) (controllerruntime.Result, error) {
+	// TODO handle VM == nil -> remove finalizer (background deletion flow)
+	// VM deletion timestamp added + VMI == nil -> remove finalizer
+	return controllerruntime.Result{}, nil
+}
+
+// Setup sets up the controller with the Manager passed in the constructor.
+func (r *VirtualMachineReconciler) Setup() error {
+	return controllerruntime.NewControllerManagedBy(r.manager).
+		For(&virtv1.VirtualMachine{}).
+		WithEventFilter(onVMPredicates()).
+		Complete(r)
+}
+
+func onVMPredicates() predicate.Funcs {
+	return predicate.Funcs{
+		CreateFunc: func(createEvent event.CreateEvent) bool {
+			return true
+		},
+		DeleteFunc: func(event.DeleteEvent) bool {
+			return true
+		},
+		UpdateFunc: func(updateEvent event.UpdateEvent) bool {
+			return false
+		},
+		GenericFunc: func(event.GenericEvent) bool {
+			return false
+		},
```

Comment:
is it fine those two are false ?
i copy pasted it from VMI controiller, and we also need update events arent we?
if so it means we need to change it there as well